llmfoundry/callbacks/inactivity_callback.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import signal
+import threading
+from typing import Optional
+
+from composer import Callback, Logger, State
+from composer.loggers import MosaicMLLogger
+
+from llmfoundry.utils.exceptions import RunTimeoutError
+
+log = logging.getLogger(__name__)
+
+
+def _timeout(timeout: int, mosaicml_logger: Optional[MosaicMLLogger] = None):
+    log.error(
+        f'Timeout after no Trainer events were triggered for {timeout} seconds.',
```

Comment:
```suggestion
        f'Timeout after {timeout} seconds of inactivity after fit_end.',
```
(since the callback only does fit end for now, can update later if we add more timeout locations)